src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -71,251 +74,372 @@ internal static int InternalConvertToUtf32(StringBuilder s, int index)
                     if ((uint)temp2 <= HIGH_SURROGATE_RANGE)
                     {
                         // Convert the surrogate to UTF32 and get the result.
-                        return (temp1 * 0x400) + temp2 + UNICODE_PLANE01_START;
+                        c = (temp1 << 18) + temp2 + UNICODE_PLANE01_START;
```

Comment:
>c = (temp1 << 18) [](start = 24, length = 17)

could you please add a small comment telling this equivalent to multiplying by 0x400?